### PR TITLE
Improved reporting of passing expectations

### PR DESCRIPTION
**Pull request by GCHQDeveloper911**
…s changes to jasmine-core (being submitted separately as required) but does not break without those changes to jasmine-core.

## Proposed changes

As above, and in line with changes made to wdio-spec-reporter and wdio-jasmine-framework, I'm looking to improve the reporting from webdriverio

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [X] I have proposed the same patch to the new [v5](https://github.com/webdriverio/v5) repository

### CODE REVIEW

File: lib/utils/BaseReporter.js [hunk 1]
```diff
@@ -208,7 +208,21 @@ class BaseReporter extends events.EventEmitter {
     }
 
     /**
-     * Outut the given failures as a list
+     * Output the given passes as a list
+     */
+    listPasses () {
+        console.log()
```
Comment (BorisOsipov) [inline lib/utils/BaseReporter.js:214]: I am not sure that any output should be in BaseReporter

**Review by BorisOsipov — CHANGES_REQUESTED** [review]
Move resutls output to particular reporter

File: lib/utils/BaseReporter.js (hunk 1 above)
Comment (GCHQDeveloper911) [reply lib/utils/BaseReporter.js:214]: I'm not disagreeing (or agreeing necessarily!) - but I created listPasses() from the pre-existing listFailures() - which has basically the same output statements? ie, listPasses works in the same way as listFailures (for consistency) - but obviously for passes not fails! Can you explain further why you feel this is wrong?

Comment (BorisOsipov) [reply lib/utils/BaseReporter.js:214]: >Can you explain further why you feel this is wrong?

I mixed v5\v4 reporter behaviour.

>but I created listPasses() from the pre-existing listFailures()

yeah I see now

File: lib/utils/ReporterStats.js [hunk 2]
```diff
@@ -95,12 +95,19 @@ class ReporterStats extends RunnableStats {
...
         this.failures = []
+        this.passes = []
     }
 
     getCounts () {
         return this.counts
     }
 
+    getPasses () {
+        return this.passes.map(function (test) {
```
Comment (BorisOsipov) [inline lib/utils/ReporterStats.js]: Why `map()` with empty callback? Maybe just `return this.passes`?

Comment (GCHQDeveloper911) [reply lib/utils/ReporterStats.js]: Fair point! Changed...

**Comment by BorisOsipov** [comment]
Close\reopen to trigger travis

**Comment by christian-bromann** [comment]
@BorisOsipov I am not if the build works. It became broken when JWT support in Travis got discontinued.

**Comment by BorisOsipov** [comment]
Ok disable Travis then for a while

**Comment by GCHQDeveloper911** [comment]
Is Travis disabled now? I tried closing/reopening just to check, but not a lot happened. Are we happy with this PR now?

**Review by BorisOsipov — APPROVED** [review]

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**